### PR TITLE
feat: download artifact

### PR DESCRIPTION
**Pull request by mzmuer**
#44 add download artifact API

### CODE REVIEW

File: pkg/artifact/artifacts_test_common.go [hunk 1]
```diff
@@ -44,3 +44,39 @@ func PrepareGetEmptyArtifacts(roundTripper *mhttp.MockRoundTripper, rootURL, use
...
 }
+
+// PrepareGetArtifact only for test
+func PrepareGetArtifact(roundTripper *mhttp.MockRoundTripper, rootURL, user, passwd, projectName, pipelineName string, buildID int, filename string) (response *http.Response) {
+	var api = fmt.Sprintf("/job/%s/job/%s/%d/artifact/%s", projectName, pipelineName, buildID, filename)
+	request, _ := http.NewRequest(http.MethodGet, fmt.Sprintf("%s%s", rootURL, api), nil)
+	response = &http.Response{
+		StatusCode: 200,
+		Request:    request,
+		Body:       ioutil.NopCloser(bytes.NewBufferString(`the is test file`)),
```
Comment (JohnNiang) [inline pkg/artifact/artifacts_test_common.go:55]:
```suggestion
		Body:       ioutil.NopCloser(bytes.NewBufferString(`this is test file`)),
```

File: pkg/artifact/artifacts_test.go [hunk 2]
```diff
@@ -57,4 +59,39 @@ var _ = Describe("artifacts test", func() {
...
+			artifactClient.Token = password
+
+			projectName := "fakename"
+			pipelineName := "fakename"
+			filename := "fakename"
+			PrepareGetArtifact(roundTripper, artifactClient.URL, username, password, projectName, pipelineName, 1, filename)
+
+			body, err := artifactClient.GetArtifact(projectName, pipelineName, 1, filename)
+			Expect(err).To(BeNil())
+			Expect(func() bool {
```
Comment (JohnNiang) [inline pkg/artifact/artifacts_test.go:75]: IMO, it would be better if we returned the body string.

**Review by JohnNiang — CHANGES_REQUESTED** [review]

File: pkg/artifact/artifacts_test.go (hunk 2 above)
Comment (mzmuer) [reply pkg/artifact/artifacts_test.go:75]: IMO, this might return a larger file. Using a body string can lead to a significant memory footprint.

Comment (LinuxSuRen) [reply pkg/artifact/artifacts_test.go:75]: agree with @mzmuer , the artifact file might be a binary file. In that case, it is not possible to return a string.

**Review by LinuxSuRen — COMMENTED** [review]
/lgtm

**Review by JohnNiang — APPROVED** [review]
LGTM

**Pull request merged**